src/sql/workbench/parts/notebook/outputs/notebookMarkdown.ts

```diff
@@ -0,0 +1,240 @@
+		const element = document.createElement(tagName);
+		if (options.className) {
+			element.className = options.className;
+		}
+		return element;
+	}
+
+	parse(text: string): any {
+		let data = JSON.parse(text);
+		data = revive(data, 0);
+		return data;
+	}
+
+	/**
+	 * Create html nodes for the given content element.
+	 * Adapted from htmlContentRenderer. Ensures that the markdown renderer
+	 * gets passed in the correct baseUrl for the notebook's saved location,
+	 * respects the trusted state of a notebook, and allows command links to
+	 * be clickable.
+	 */
+	renderMarkdown(markdown: IMarkdownString, options: RenderOptions = {}): HTMLElement {
+		const element = this.createElement(options);
+
+		// signal to code-block render that the element has been created
+		let signalInnerHTML: () => void;
+		const withInnerHTML = new Promise(c => signalInnerHTML = c);
+
+		let notebookFolder = path.join(path.dirname(this._notebookURI.fsPath), path.sep);
+		if (!this._baseUrls.includes(notebookFolder)) {
+			this._baseUrls.push(notebookFolder);
+		}
+		const renderer = new marked.Renderer({ baseUrl: notebookFolder });
+		renderer.image = (href: string, title: string, text: string) => {
+			href = this.cleanUrl(!markdown.isTrusted, notebookFolder, href);
+			let dimensions: string[] = [];
+			if (href) {
+				const splitted = href.split('|').map(s => s.trim());
+				href = splitted[0];
+				const parameters = splitted[1];
+				if (parameters) {
+					const heightFromParams = /height=(\d+)/.exec(parameters);
+					const widthFromParams = /width=(\d+)/.exec(parameters);
+					const height = heightFromParams ? heightFromParams[1] : '';
+					const width = widthFromParams ? widthFromParams[1] : '';
+					const widthIsFinite = isFinite(parseInt(width));
+					const heightIsFinite = isFinite(parseInt(height));
+					if (widthIsFinite) {
+						dimensions.push(`width="${width}"`);
+					}
+					if (heightIsFinite) {
+						dimensions.push(`height="${height}"`);
+					}
+				}
+			}
+			let attributes: string[] = [];
+			if (href) {
+				attributes.push(`src="${href}"`);
+			}
+			if (text) {
+				attributes.push(`alt="${text}"`);
+			}
+			if (title) {
+				attributes.push(`title="${title}"`);
+			}
+			if (dimensions.length) {
+				attributes = attributes.concat(dimensions);
+			}
+			return '<img ' + attributes.join(' ') + '>';
+		};
+		renderer.link = (href, title, text): string => {
+			href = this.cleanUrl(!markdown.isTrusted, notebookFolder, href);
+			if (href === null) {
+				return text;
+			}
+			// Remove markdown escapes. Workaround for https://github.com/chjj/marked/issues/829
+			if (href === text) { // raw link case
+				text = removeMarkdownEscapes(text);
+			}
+			title = removeMarkdownEscapes(title);
+			href = removeMarkdownEscapes(href);
+			if (
+				!href
+				|| !markdown.isTrusted
+				|| href.match(/^data:|javascript:/i)
+				|| href.match(/^command:(\/\/\/)?_workbench\.downloadResource/i)
```

Comment:
Is this removing support for command syntax? What specifically are these 2 lines blocking? I just want to make sure the feature where, when trusted, we can execute commands is still present. Thanks!

Reply:
I tested commands and they work just fine when trusted. I think this just prevents workbench.downloadResource (which I'm assuming was done out of security concerns).

Reply:
The second line is blocking pure javascript. Since we don't have a renderer any preference over whether we should get rid of this block here?

Reply:
This is fine to keep as-is